app/lib/actions/katello/host/update_system_purpose.rb

```diff
@@ -0,0 +1,28 @@
+module Actions
+  module Katello
+    module Host
+      class UpdateSystemPurpose < Actions::EntryAction
+        def plan(host, service_level, purpose_role, purpose_usage)
+          if host.content_facet
```

Comment:
Probably a bit more precise to check if the subscription_facet exists rather than content_facet, but I guess it works.

Style comment: you could do this on one line if you want: `fail ("") unless host.subscription_facet

Reply:
Updated and made it cleaner